src/api/api.ts

```diff
@@ -0,0 +1,36 @@
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export const getUsers = () => {
+  return client.get<User[]>('/users');
+};
+
+export const getPosts = ({ id }: Pick<User, 'id'>) => {
+  return client.get<Post[]>(`/posts?userId=${id}`);
+};
+
+export const getComments = ({ id }: Pick<Post, 'id'>) => {
+  return client.get<Comment[]>(`/comments?postId=${id}`);
```

Comment:
Add error handling for the `getComments` function to ensure that any errors during the API call are properly managed.